packages/config/src/handlers/locations/children.ts

```diff
@@ -10,19 +10,25 @@
  */
 import { UUID } from '@opencrvs/commons'
 import { ServerRoute } from '@hapi/hapi'
+import { fetchFromHearth } from '@config/services/hearth'
+import { SavedLocation } from '@opencrvs/commons/types'
 import { resolveLocationChildren } from './locationTreeSolver'
-import { fetchLocations } from '@config/services/hearth'
-import { find } from 'lodash'
-import { notFound } from '@hapi/boom'
 
 export const resolveChildren: ServerRoute['handler'] = async (req) => {
   const { locationId } = req.params as { locationId: UUID }
-  const locations = await fetchLocations()
-  const location = find(locations, { id: locationId })
 
-  if (!location) {
-    return notFound()
+  const location = await fetchFromHearth<SavedLocation>(
+    `Location/${locationId}`
+  )
+  if (isTypeOf(location, 'CRVS_OFFICE')) {
+    return [location]
   }
 
-  return [location, ...resolveLocationChildren(location, locations)]
+  const children = await resolveLocationChildren(locationId)
+
+  return [location, ...children]
+}
+
+function isTypeOf(location: SavedLocation, type: string) {
+  return location?.type?.coding?.some((x) => x.code === type)
 }
```

Comment:
https://github.com/opencrvs/opencrvs-core/blob/16759a9c43dff8ba3d002ffb7b71e6ad7282afdb/packages/commons/src/fhir/location.ts#L118
We have this type of `isHealthFacility` helper in commons, a `isCrvsOffice` could be good for consistency?